ModsenOnlineStore.Store.Infrastructure/Services/OrderService/OrderService.cs

```diff
@@ -0,0 +1,84 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using ModsenOnlineStore.Common;
+using ModsenOnlineStore.Store.Application.Interfaces.OrderInterfaces;
+using ModsenOnlineStore.Store.Domain.DTOs.OrderDTOs;
+using ModsenOnlineStore.Store.Domain.Entities;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ModsenOnlineStore.Store.Infrastructure.Services.OrderService
+{
+        public class OrderService : IOrderService
+        {
+            private IMapper mapper;
+            private IOrderRepository repository;
+
+            public OrderService(IMapper mapper, IOrderRepository repository)
+            {
+                this.mapper = mapper;
+                this.repository = repository;
+            }
+
+            public async Task<ResponseInfo<List<GetOrderDTO>>> GetAllOrders()
+            {
+                var orders = await repository.GetAllOrders();
+                var orderDTOs = orders.Select(p => mapper.Map<GetOrderDTO>(p)).ToList();
+                if (orderDTOs.Count == 0)
+                {
+                    return new ResponseInfo<List<GetOrderDTO>>(orderDTOs, true, "no orders yet");
+                }
+                return new ResponseInfo<List<GetOrderDTO>>(orderDTOs, true, "all orders");
+            }
+
+            public async Task<ResponseInfo<GetOrderDTO>> GetSingleOrder(int id)
+            {
+                var order = await repository.GetSingleOrder(id);
+                if (order is null)
+                {
+                    return new ResponseInfo<GetOrderDTO>(null, false, "order not found");
+                }
+                return new ResponseInfo<GetOrderDTO>(mapper.Map<GetOrderDTO>(order), true, $"order with id {id}");
+            }
+
+            public async Task<ResponseInfo<string>> AddOrder(AddOrderDTO addOrder)
+            {
+                var newOrder = mapper.Map<Order>(addOrder);
+                await repository.AddOrder(newOrder);
+                //return await GetAllOrders();
+                return new ResponseInfo<string>(data: "added successfully", success: true, message: "order");
```

Comment:
Define what do you set in `data` arg
check for all methods

Reply:
still actual
![image](https://github.com/Gromiko-Evgeniy/ModsenOnlineStore/assets/117982737/048e7796-0c13-4760-980a-bd053a72b69a)
as you can see, the first argument is for data it self - not for message
please, synchronize your response models (name, properties and other members must be the same)
screenshot from [Pull request 8](https://github.com/Gromiko-Evgeniy/ModsenOnlineStore/pull/8/files#diff-a68b40b4cd5d4f796d15071da5dc2b02fd707168e95a65cb3864abfac0902a09)